marker_api/src/lint.rs

```diff
@@ -35,13 +57,11 @@ pub struct Lint {
     ///
     /// See [`MacroReport`] for the possible levels.
     pub report_in_macro: MacroReport,
-    // FIXME: We might want to add more fields. This should be possible as this
-    // struct is always constructed by a macro controlled by marker. These are some
-    // additional fields used  in rustc:
-    // * pub edition_lint_opts: Option<(Edition, Level)>,
-    // * pub future_incompatible: Option<FutureIncompatibleInfo>,
-    // * pub feature_gate: Option<&'static str>,
-    // * pub crate_level_only: bool,
+
+    /// This struct should always be instantiated using the [`declare_lint`](crate::declare_lint)
+    /// macro. This value is simply here, to force any construction to acknowledge the
+    /// instability of manual construction.
+    pub _unstable_i_accept_the_risk_of_instability: u8,
```

Comment:
Let's make this zero cost :rocket: 
```suggestion
    pub _unstable_i_accept_the_risk_of_instability: (),
```

Reply:
I believe that `()` inside `#[repr(C)]` structs are undefined behavior. A `&()` reference or pointer, like Marker uses, is fine, but the `()` type isn't AFAIK. Otherwise, this would be the way to go.

Reply:
I don't think that `()` is UB in `#[repr(C)]`. If it was so, then `rustc` would definitely not allow using ZST in `#[repr(C)]`, because the paradigm of `rustc` is that everything that may trigger a UB must be explicitly annotated with `unsafe` or otherwise a compile error will be emitted.

According to [rustonomicon](https://doc.rust-lang.org/nomicon/other-reprs.html) ZST are allowed in `#[repr(C)]` and they still ZST (counterintuitive to C's requirement that empty struct must take up 1 byte of memory).

Reply:
Ah okay, then it was just an empty struct, with only one `_: ()` field in it, that caused a warning. I'll update it :)